db/mongo/client.go

```diff
@@ -0,0 +1,136 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	wraperrors "github.com/pkg/errors"
+	guuid "github.com/satori/go.uuid"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github/wasimkhan042/uStore-Auth/config"
+	"github/wasimkhan042/uStore-Auth/db"
+	"github/wasimkhan042/uStore-Auth/models"
+)
+
+const (
+	userCollection  = "user"
+	subscriptionCollection = "subscription"
+)
+
+func init() {
+	db.Register("mongo", NewMongoClient)
+}
+
+type client struct {
+	conn *mongo.Client
+}
+
+// NewMongoClient initializes a mongo database connection.
+func NewMongoClient(conf db.Option) (db.DataStore, error) {
+	uri := fmt.Sprintf("mongodb://%s:%s", viper.GetString(config.DbHost), viper.GetString(config.DbPort))
+	log().Infof("initializing mongodb: %s", uri)
+
+	cli, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+	if err != nil {
+		return nil, wraperrors.Wrap(err, "failed to connect")
+	}
+
+	return &client{conn: cli}, nil
+}
+
+// Registration stores new user details in 'user' collection
+func (m client) Registration(userInfo *models.User) error {
+	_, err := m.ViewProfile(userInfo.Email)
+	// add unregistered users only
+	if err != nil {
+		if userInfo.ID != "" {
+			return nil
```

Comment:
do we really need this?

Reply:
not really... followed moeid boss repo... As per the testcases it is required but not actually because i wouldn't be asking user to enter his id.
Yes, If I want user to enter his CNIC number as his ID number...